### PR TITLE
Error 500 issue resolved for Discovery View click

### PR DESCRIPTION
**Pull request by rmathew1011**
Error 500 resolved for Discovery View click

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ spring:
     generate-ddl: true
     show-sql: false
     hibernate:
-      ddl-auto: create-drop
+      ddl-auto: update
```
Comment (wwelling) [inline src/main/resources/application.yml:53]: Could we please revert the default application.yml changes?

File: src/main/java/edu/tamu/sage/controller/DiscoveryViewController.java [hunk 2]
```diff
@@ -104,8 +105,12 @@ public ApiResponse findBySlug(@PathVariable String slug, @RequestParam(name = "f
                 PageRequest pageRequest = PageRequest.of(page.getPageNumber(), page.getPageSize(),Sort.by(defaultDirection, field));
                 page = pageRequest;
             } else {
-                String sort = page.getSort().toString().split(":")[0];
-                page = PageRequest.of(page.getPageNumber(), page.getPageSize(),Sort.Direction.fromString(direction), sort);
+                String sort = (page.getSort().toString().split(":")[0] != "UNSORTED") ? page.getSort().toString().split(":")[0] : discoveryView.getResourceLocationUriKey().replaceAll("[{}]", "");
```
Comment (wwelling) [inline src/main/java/edu/tamu/sage/controller/DiscoveryViewController.java:108]: String comparison should use equals method. The split operation only needs to happen once.  The default direction of the discovery view should be respected. Some additional suggested cleanup.

https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/EnumUtils.html#isValidEnum-java.lang.Class-java.lang.String-

```
Direction defaultDirection = discoveryView.isAscending() ? Direction.ASC : Direction.DESC;
...

String sort = page.getSort().toString().split(":")[0];
sort = !sort.equals("UNSORTED") ? sort : discoveryView.getResourceLocationUriKey().replaceAll("[{}]", "");
Direction dir = EnumUtils.isValidEnum(Direction.class, direction) ? Direction.valueOf(direction) : defaultDirection ;
page = PageRequest.of(page.getPageNumber(), page.getPageSize(), dir, sort);
```

**Review by wwelling — CHANGES_REQUESTED** [review]
String comparison should use equals method. The split operation only needs to happen once. The default direction of the discovery view should be respected.

File: src/main/webapp/app/config/appConfig.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ var appConfig = {
   'allowAnonymous': true,
   'anonymousRole': 'ROLE_ANONYMOUS',
 
-  'authStrategies': ['emailRegistration'],
```
Comment (wwelling) [inline src/main/webapp/app/config/appConfig.js:8]: Could we please revert the default appConfig.js changes?

File: src/main/resources/application.yml (hunk 1 above)
Comment (rmathew1011) [reply src/main/resources/application.yml:53]: Reverted

File: src/main/webapp/app/config/appConfig.js (hunk 3 above)
Comment (rmathew1011) [reply src/main/webapp/app/config/appConfig.js:8]: Reverted

File: src/main/java/edu/tamu/sage/controller/DiscoveryViewController.java (hunk 2 above)
Comment (rmathew1011) [reply src/main/java/edu/tamu/sage/controller/DiscoveryViewController.java:108]: Utilized this block

**Review by wwelling — APPROVED** [review]

**Pull request merged**